.github/workflows/bump-version.yml

```diff
@@ -1,44 +1,106 @@
+name: Bump Version
+
 on:
-    pull_request:
-      branches:
-        - main
-      types: [closed]
-  
+  pull_request:
+    types: [closed]
+    branches: [main]
+
 jobs:
-    bump-version:
-      runs-on: ubuntu-latest
-  
-      env:
-        GH_PUSH_TOKEN: ${{ secrets.GH_PUSH_TOKEN }}
-  
-      steps:
-  
-        - name: Checkout
-          uses: actions/checkout@v4
-          if: env.GH_PUSH_TOKEN != ''
-          with:
-            token: ${{ env.GH_PUSH_TOKEN }}
-  
-        - name: configure git
-          if: env.GH_PUSH_TOKEN != ''
-          run: |
-            git remote set-url origin https://itsalaidbacklife@github.com/cuttle-cards/cuttle
-            git config --global user.name "Cuttle Continuous Integration"
-            git config --global user.email "itsalaidbacklife@gmail.com"
-  
-  
-        - name: bump patch version
-          if: contains(github.event.pull_request.labels.*.name, 'version-patch') && env.GH_PUSH_TOKEN != ''
-          run: npm run version:patch
-  
-        - name: bump minor version
-          if: contains(github.event.pull_request.labels.*.name, 'version-minor') && env.GH_PUSH_TOKEN != ''
-          run: npm run version:minor
-  
-        - name: bump major version
-          if: contains(github.event.pull_request.labels.*.name, 'version-major') && env.GH_PUSH_TOKEN != ''
-          run: npm run version:major
-  
-        - name: push
-          if: env.GH_PUSH_TOKEN != ''
-          run: git push origin main
+  check-token:
+    runs-on: ubuntu-latest
+    outputs:
+      GH_PUSH_TOKEN: ${{ steps.check.outputs.GH_PUSH_TOKEN }}
+    steps:
+      - name: Check GH_PUSH_TOKEN
+        id: check
+        run: |
+          if [[ -z "${{ secrets.GH_PUSH_TOKEN }}" ]]; then
+            echo "GH_PUSH_TOKEN is not set. Exiting."
+            exit 1
+          fi
+          echo "::set-output name=GH_PUSH_TOKEN::${{ secrets.GH_PUSH_TOKEN }}"
+        shell: bash
+
+  check-merged:
+    needs: check-token
+    runs-on: ubuntu-latest
+    outputs:
+      merged: ${{ steps.check.outputs.merged }}
+    steps:
+      - name: Check if PR is merged
+        id: check
+        run: |
+          if [[ "${{ github.event.pull_request.merged }}" != "true" ]]; then
+            echo "Pull request is not merged. Exiting."
+            exit 1
+          fi
+          echo "::set-output name=merged::${{ github.event.pull_request.merged }}"
+        shell: bash
+
+  get-labels:
```

Comment:
4. **Get Labels:** The `get-labels` job runs next, contingent on the successful completion of `check-merged`. It retrieves the labels from the pull request and verifies if exactly one of them matches the pattern `version-(patch|minor|major)`. If not, the job fails and the workflow terminates. If it does match, the job outputs the version label for subsequent jobs.